Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -0,0 +1,457 @@
+            //    long.
+            let (connectionID, eventLoop) = self.connections.parkConnection(at: index)
+            return .init(
+                request: .none,
+                connection: .scheduleTimeoutTimer(connectionID, on: eventLoop)
+            )
+        }
+
+        private mutating func nextActionForIdleEventLoopConnection(
+            at index: Int,
+            context: HTTP1Connections.IdleConnectionContext
+        ) -> Action {
+            // Check if there are waiting requests in the matching eventLoop queue
+            if let request = self.queue.popFirst(for: context.eventLoop) {
+                return .init(
+                    request: .executeRequest(request, self.connections.leaseConnection(at: index), cancelTimeout: true),
+                    connection: .none
+                )
+            }
+
+            // TBD: What do we want to do, if there are more requests in the general purpose queue?
+            //      For now, we don't care. The general purpose connections will pick those up
+            //      eventually.
+            //
+            // If there is no more eventLoop bound work, we close the eventLoop bound connections.
+            // We don't park them.
+            return .init(
+                request: .none,
+                connection: .closeConnection(self.connections.closeConnection(at: index), isShutdown: .no)
+            )
+        }
+
+        // MARK: Failed/Closed connection management
+
+        private mutating func nextActionForFailedConnection(
+            at index: Int,
+            context: HTTP1Connections.FailedConnectionContext
+        ) -> Action {
+            switch self.state {
+            case .running:
+                switch context.use {
+                case .generalPurpose:
+                    return self.nextActionForFailedGeneralPurposeConnection(at: index, context: context)
+                case .eventLoop:
+                    return self.nextActionForFailedEventLoopConnection(at: index, context: context)
+                }
+
+            case .shuttingDown(let unclean):
+                assert(self.queue.isEmpty)
+                self.connections.removeConnection(at: index)
+                if self.connections.isEmpty {
+                    return .init(
+                        request: .none,
+                        connection: .cleanupConnections(.init(), isShutdown: .yes(unclean: unclean))
+                    )
+                }
+                return .none
+
+            case .shutDown:
+                preconditionFailure("If the pool is already shutdown, all connections must have been torn down.")
+            }
+        }
+
+        private mutating func nextActionForFailedGeneralPurposeConnection(
+            at index: Int,
+            context: HTTP1Connections.FailedConnectionContext
+        ) -> Action {
+            if context.connectionsStartingForUseCase < self.queue.count(for: nil) {
```

Comment:
Perhaps file an issue for future enhancement (no sense gluing it into this PR), but I don't love this spelling. Probably `self.queue.generalPurposeCount` would be clearer.